luaui/Widgets/unit_cloak_firestate.lua

```diff
@@ -58,6 +58,19 @@ end
 
 local decloakFireState = {} --stores the desired fire state when decloaked of each unitID
 
+local function GiveNotifyingOrderToUnit(uID, cmdID, cmdParams, cmdOpts)
+	-- Other widgets interact with the fire state, so we need to notify them when issuing the order. This affects the holdfire fix and when cloak "kicks in".
+	for _, w in ipairs(widgetHandler.widgets) do
+		if w.UnitCommandNotify and w:UnitCommandNotify(uID, cmdID, cmdParams, cmdOpts) then
+			-- Per Spring, if Notify returns true, it deletes the command.
+			return
+		end
+	end
```

Comment:
```suggestion
	if Script.LuaUI.UnitCommandNotify then
		Script.LuaUI.UnitCommandNotify(uID, cmdID, cmdParams, cmdOpts)
	end
```
@sprunk This is the proper way to manually invoke callins?

Reply:
Check if `Script.LuaUI.Foo` returns the bool, there's been some issues around that. If yes then I'd go with the `Script.LuaUI` route.

Reply:
It doesn't return the bool, it just returns nil.

Reply:
The actual callin to check is probably just `CommandNotify` and not `UnitCommandNotify`

Reply:
Don't see much consistency if everyone actually uses CommandNotify instead of UnitCommandNotify. Also don't see UnitCommandNotify even exists, that's why I was asking.

So if you want to have CommandNotify block the command you can use widgetHandler:CommandNotify directly since you set `handler = true`. Still not sure about this, since there are other ways gadgets or widgets can block commands for example with AllowCommand, so pbbly best is just letting the engine do its thing with the command afaics.

A different solution should probably be found that doesn't involve two different CommandNotify callins to get that stop command to be inserted, maybe Script.LuaUI to notify the other widget would be best, to avoid duplication and let it do its thing.

Reply:
Actually, thinking about this, to not overcomplicate the issue and keep it simple I think best will be smth like:

```lua
if not widgetHandler:CommandNotify(...) then
   Script.LuaUI.CloakHoldFire(uID, cmdID, cmdParams, cmdOpts) -- or LuaUI.HoldFire or smth.
end
```

The way how `cmd_holdfire_fix.lua` does it doesn't seem like a good way to insert the stop, since then it will insert it even if a widget later in the layer priority actually cancels the holdfire. I guess there's no problem for now since probably nobody cancelling it. Also it's not in scope of this PR so maybe don't worry about it, you could fix that in another PR if you feel like dedicating the time, otherwise it's just a "latent bug".